http-api/src/v1/projects.rs

```diff
@@ -106,44 +100,23 @@ async fn get_projects_info(
     let per_page = per_page.unwrap_or(10);
 
     let storage = ctx.storage().await?;
-    let projects: Vec<Project> = identities::any::list(storage.read_only())?
-        .map(|res| match res {
-            Ok(id) => Ok(id),
-            Err(err) => Err(Error::from(err)),
-        })
+    let projects = identities::any::list(storage.read_only())?
         .filter_map(|id_result| match id_result {
-            Ok(SomeIdentity::Project(project)) => Some(Ok(project)),
-            Err(err) => Some(Err(err)),
+            Ok(SomeIdentity::Project(project)) => {
+                if let Ok(info) = get_project_info_sync(ctx.clone(), project, storage.as_ref()) {
```

Comment:
Hey, you have removed the `spawn_blocking()` stuff.  `get_project_info_sync()` is a pretty expensive operation (and a blocking one) and it tends to block processing of all HTTP requests if called in a blocking manner.  `spawn_blocking()` is there to keep on processing other requests while we wait for results of `get_project_info_sync()`

Reply:
Ok, that makes sense.. Would it be possible to block the iterator somehow to wait for results of `get_project_info_sync()` so we can paginate over a collection of valid projects?

Reply:
I’m not sure what you mean by blocking the iterator exactly but one way fix the issue is to wrap the entire logic (with the `spawn_blocking()` and `join_all()`) in a `loop {}` that breaks when a sufficient number of project infos has been collected (or there isn’t any more projects left).  In case it’s not clear, any iterator can be advanced one step simply by calling its `next()` method:

```Rust
loop {
  let res = match identities::any::list(storage.read_only())? {
    Some(res) => res,
    None => break,
  };
  […]
}
```